app/models/reward.rb

```diff
@@ -15,4 +16,14 @@ class Reward < ApplicationRecord
   def display_category
     category.present? ? category.title : 'without category'
   end
+
+  def number_of_available_items
+    delivery_method_online? ? online_codes.where(status: :not_used).count : 'undetermined'
+  end
+
+  def available_for_purchase?
+    return true if delivery_method_post_delivery? || number_of_available_items.positive?
```

Comment:
Understood :) Honestly I thought about it when I was implementing these methods. I summarized that this kind of error would not occur because `number_of_available_items` returns string only when we have post delivery. The `available_for_purchase?` method first checks if reward has post delivery and (if it's true) returns true. So I assumed that there is no option for calling `.positive? method on string.

I also see the argument that a method should return only one type of value 👍  Currently acceptance criterias don't mention about manner of couting the number of available items when delivery method is post delivery. Thus I decided to return `0` insted of `undetermined` string. At now `available_for_purchase?` method returns true directly when a reward has post delivery so it shouldn't be a problem. :)

Reply:
The construct which you had created would work as you intended but lacks readability as it's hard to reveal your intention. It's better when a reader doesn't have to double read a piece of code to get how it works and why. A single type of returned value makes its usage much cleaner and having less conditional branching.

You can probably reduce this method to
```ruby
def available_for_purchase?
  delivery_method_post_delivery? || number_of_available_items.positive?
end
```

:)

Reply:
Understood! Thanks :) Of course I changed syntax according to your suggestion 👍